src/gui/optionsdialog.cpp

```diff
@@ -1165,6 +1167,8 @@ void OptionsDialog::loadOptions()
     // Security
     m_ui->checkClickjacking->setChecked(pref->isWebUiClickjackingProtectionEnabled());
     m_ui->checkCSRFProtection->setChecked(pref->isWebUiCSRFProtectionEnabled());
+    m_ui->checkSecureCookie->setEnabled(pref->isWebUiHttpsEnabled());
```

Comment:
You can do it better by disabling the checkbox in real-time, not just when the dialog loads. An example for you to find out:
`connect(m_ui->checkExportDir, &QAbstractButton::toggled, m_ui->textExportDir, &QWidget::setEnabled);`

Reply:
Sorry, can't see it now. Aren't all HTTPS related settings under some group box? Then we can just enable/disable it.

Reply:
@glassez I saw that many related settings are in the same `QGroupBox`, which gives you the disabling logic "for free". However, this is not the case for this setting, as it is in a different group ("Security") from the HTTPS setting, and I didn't think it made sense to move it.

Reply:
@Chocobo1 
Something like this?

```cpp
connect(m_ui->checkWebUiHttps, &QGroupBox::toggled, m_ui->checkSecureCookie, &QWidget::setEnabled);
```

And with this, do I no longer need that line above?

Reply:
>and I didn't think it made sense to move it.

I feel it as related to HTTPS.
No matter...

Reply:
@Chocobo1 changes implemented, disregard question above, I figured out I also needed to set the checkbox state on options load.

@glassez yes, it is _related_ to HTTPS but as it is a _security_ setting so it is grouped with the other security settings. I think there is no 100% ideal place for it, as there is a valid argument for both sides, but I think this is the most correct one.

Reply:
I just thought it has no meaning without HTTPS...
Really HTTPS is part of "Security" options. Maybe we should move it there? And then move "security cookie" under HTTPS so it will under Security.HTTPS.
Just thinking out loud...

Reply:
Personally I think WebUI/WebAPI users should care about every option in the WebUI tab except the ones in Security section. The Security section shouldn't be bother with without a good understanding about them and I think this secure cookie flag belongs to it just fine.